### ISSUE TITLE
feature: implement `toHaveAccessibleValue` Jest matcher

### ISSUE THREAD

**Issue by mdjastrzebski** [callstack/react-native-testing-library#1485]
## Describe the Feature

[`toHaveAccessibleName`](https://github.com/testing-library/jest-dom#tohaveaccessiblename) is a Jest DOM matcher asserting that given element has certain accessible name: text content or a11y label.

RNTL already contains some code working with accessible name, namely `*ByRole` query accepts an optional [`name` param](https://github.com/callstack/react-native-testing-library/blob/a9f32a69e46bedc9413dd0bdf74637f7a3b2b579/src/queries/role.ts#L37). This matcher should compare expect name parameter with all of these:
1. text content of element referenced by `accessibilityLabelledBy` / `aria-labelledby` prop
2. `accessbilityLabel` / `aria-label` prop
3. text content of given element

These should be evaluated in that particular order.

## Possible Implementations
Proposed API:

```ts
export function toHaveAccessibleName(
  this: jest.MatcherContext,
  element: ReactTestInstance,
  expectedName?: TextMatch,
) 
```

The matcher should:
1) Validate that it is invoked on a host element
2) When `expectedName` param is provided check if it matches element's accessible name
3) When `expectedName` is not provided check if element has accessible name

Each matcher should have a fairly comprehensive test suite.

## Related Issues
#1454

**Comment by tarunrajput** [comment]
@mdjastrzebski, I'd like to implement this

**Comment by mdjastrzebski** [comment]
@tarunrajput go ahead, it's yours now!

**Comment by mdjastrzebski** [comment]
Resolved in #1496!

**Comment by mdjastrzebski** [comment]
🎉 Release in version: [v12.4.0](https://github.com/callstack/react-native-testing-library/releases/tag/v12.4.0)
